### PR TITLE
Fix condition check for chunked transfers

### LINKED ISSUE #1019: HLS stream AES-128 encryption
## Bug report
Hi I was wondering if you could help me out or point me in right direction, as last couple days I'm kicking with the horse
### Describe the bug
`2022-07-13 01:12:53.039 T:8696    FATAL <general>: AddOnLog: inputstream.adaptive: AES decryption failed: -10`

Here is a clear and concise description of what the problem is:

Website which I'm cooperating started implementing new player, during tests performed there was no problem.
However during tests requests for key always returned 16bytes 0. `b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'`
Recently thats changed and key request returning values (16 bytes). 
Different everytime
Best I've got is 3seconds playback  and spamming decryption failed in log.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

## Possible Fix

Any advice - point appreciated

### To Reproduce

1. Script required some workaround to get hls playlist possible to run,  also there is proxy set up
   as server require specific ping to keep stream active. 
I can share it if required, not sure if it needed now.
2. Example link: [https://iframe.mediadelivery.net/embed/43012/7e5a66c8-7d0a-4f05-865b-2b76e549748a](url)
But as mention before it need workaround to get hls manifest and required headers.

Let me know - will provide more details when needed

### Debuglog

The debuglog can be found here:
[https://paste.kodi.tv/letexegeje.kodi](url)

### MPD/M3U8s/ISM

An example or copy of a manifest (or manifests for HLS - master and variants) can be found here:
```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-TARGETDURATION:5
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-PLAYLIST-TYPE:VOD
#EXT-X-KEY:METHOD=AES-128,URI="https://video-420.mediadelivery.net/.drm/063cd965-66a8-4fe6-8509-681c6b748a4f/1280x720.drmkey?v=0",IV=0x3230386236373936636331332d346433
#EXTINF:5.005000,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video0.ts?v=0&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:3.753756,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video1.ts?v=0&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:3.753744,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video2.ts?v=0&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXT-X-KEY:METHOD=AES-128,URI="https://video-420.mediadelivery.net/.drm/063cd965-66a8-4fe6-8509-681c6b748a4f/1280x720.drmkey?v=1",IV=0x3230386236373936636331332d346433
#EXTINF:3.753756,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video3.ts?v=1&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:3.753744,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video4.ts?v=1&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:5.005000,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video5.ts?v=1&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXT-X-KEY:METHOD=AES-128,URI="https://video-420.mediadelivery.net/.drm/063cd965-66a8-4fe6-8509-681c6b748a4f/1280x720.drmkey?v=2",IV=0x3230386236373936636331332d346433
#EXTINF:3.753756,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video6.ts?v=2&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:3.753744,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video7.ts?v=2&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXTINF:3.753756,
https://vz-86fc79ca-5ba.b-cdn.net/7e5a66c8-7d0a-4f05-865b-2b76e549748a/1280x720/video8.ts?v=2&resolution=1280x720&server=420&contextId=063cd965-66a8-4fe6-8509-681c6b748a4f
#EXT-X-KEY:METHOD=AES-128,URI="https://video-420.mediadelivery.net/.drm/063cd965-66a8-4fe6-8509-681c6b748a4f/1280x720.drmkey?v=3",IV=0x3230386236373936636331332d346433
```

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [x] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [x] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version: 19.3
 - inputstream.adaptive version: 19.0.7

*note: Once the issue is made we require you to update it with new information should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by glennguy**
For the case of HTTP2 where `Transfer-Encoding: Chunked` is not a thing. Now `lastChunk` is only set on HTTP1 when a content length is specified.
To fix #1019 

@Lantash77 could you try the build for Kodi 20 [here](https://jenkins.kodi.tv/blue/organizations/jenkins/xbmc%2Finputstream.adaptive/detail/PR-1022/1/artifacts)?

### CODE REVIEW

**Comment by Lantash77** [comment]
#1019 
Confirming
Solving problem. 
Thank you

File: src/common/AdaptiveStream.cpp [hunk 1]
```diff
@@ -230,7 +230,10 @@ bool AdaptiveStream::download(const std::string& url,
     bool isEOF{false};
     std::string transferEncodingStr{
         file.GetPropertyValue(ADDON_FILE_PROPERTY_RESPONSE_HEADER, "Transfer-Encoding")};
-    bool isChunked{transferEncodingStr.find("hunked") != std::string::npos};
+    std::string contentLengthStr{
+            file.GetPropertyValue(ADDON_FILE_PROPERTY_RESPONSE_HEADER, "Content-Length") };
+    bool isChunked{contentLengthStr.empty() ||
```
Comment (CastagnaIT) [inline src/common/AdaptiveStream.cpp:237]: Can you add a little comment that explain why has been done in this way?
something like:
HTTP2 content is always chunked and Content-Length header should not be provided

Comment (CastagnaIT) [reply src/common/AdaptiveStream.cpp:237]: please reapply clang

Comment (glennguy) [reply src/common/AdaptiveStream.cpp:237]: Thanks, I have added some detailed comments

**Review by CastagnaIT — APPROVED** [review]

**Pull request merged**